src/psyclone/tests/test_files/dynamo0p3/16.2_integer_scalar_sum.f90

```diff
@@ -11,6 +12,6 @@ program single_invoke
   integer(i_def)   :: isum
   type(field_type) :: f1
 
-  call invoke( inner_product(isum, f1) )
+  call invoke( X_innerproduct_Y(isum, f1) )
```

Comment:
The number of arguments passed to X_innerproduct_Y is wrong. Please could you elaborate on my original comment (which I now realise is too brief) and explain that this test is to be used with incorrect meta-data for X_innerproduct_Y such that it claims to perform a reduction into an integer variable.

Reply:
Corrected the number of arguments and updated the description. I also updated the description for this test in dynamo0p3_builtins_test.py as it did not completely reflect the test's functionality. 
It tests for a reduction into an integer scalar rather than an integer argument per se.